prestashopv1/plugin/modules/indodana/controllers/front/validation.php

```diff
@@ -0,0 +1,158 @@
+        }
+
+        // Check that this payment option is still available in case the
+        // customer changed his address just before the end of the checkout process
+        $authorized = false;
+        foreach (Module::getPaymentModules() as $module) {
+            if ($module['name'] == 'indodana') {
+                $authorized = true;
+                break;
+            }
+        }
+        if (!$authorized) {
+            die($this->module->getTranslator()->trans('This payment method is not available.', [], 'Modules.Wirepayment.Shop'));
+        }
+
+        $customer = new Customer($cart->id_customer);
+        if (!Validate::isLoadedObject($customer)) {
+            Tools::redirect('index.php?controller=order&step=1');
+        }
+
+        $currency = $this->context->currency;
+        $total = (float) $cart->getOrderTotal(true, Cart::BOTH);
+
+        if ($this->isValidOrder() === true) {
+            $paymentStatus = Configuration::get('INDODANA_DEFAULT_ORDER_PENDING_STATUS');
+            $message = null;
+        } else {
+            $paymentStatus = Configuration::get('PS_OS_ERROR');
+
+            /**
+             * Add a message to explain why the order has not been validated
+             */
+            $message = $this->module->l('An error occurred while processing payment');
+        }
+
+        $this->module->validateOrder(
+            $cart->id,
+            $paymentStatus,
+            $total,
+            $this->module->displayName,
+            $message,
+            [],
+            (int) $currency->id,
+            false,
+            $customer->secure_key
+        );
+
+        $orderId = Order::getOrderByCartId((int) $cart->id);
+        $orderDetails = new Order((int) $orderId);
+
+        /**
+         * An error occured and is shown on a new page.
+         */
+        if (!$orderId) {
+            $this->errors[] = $this->module->l('An error occured. Please contact the merchant to have more informations');
+
+            return $this->setTemplate('error.tpl');
+        }
+
+
+        /**
+         * The order has been placed so we redirect the customer on the indodana payment page.
+         */
+        $moduleId = $this->module->id;
+        $approveUrl = $this->context->link->getModuleLink(
+            $this->module->name,
+            'approve',
+            ['id_order' => $orderId],
+            true
+        );
+        $cancelUrl = $this->context->link->getModuleLink(
+            $this->module->name,
+            'cancel',
+            ['id_order' => $orderId],
+            true
+        );
+        $lang = $this->context->language->iso_code;
+        $backUrl = _PS_BASE_URL_ . '/' . $lang  . '/order-confirmation?id_cart=' . $cart->id . '&id_module=' . $moduleId . '&id_order=' . $orderId . '&key=' . $customer->secure_key;
+
+        $checkoutUrl = IndodanaTools::getIndodanaCommon()->checkout([
+            'merchantOrderId' => $orderId . '-' . $orderDetails->reference,
```

Comment:
What are the example end values of `merchantOrderId`?

Reply:
orderId = 1
reference = XKBKNABJK (random string generated by Prestashop)
merchantOrderId = 1-XKBKNABJK